README.md

```diff
@@ -473,7 +473,14 @@ instance.defaults.headers.common['Authorization'] = AUTH_TOKEN;
 
 ### Config order of precedence
 
-Config will be merged with an order of precedence. The order is library defaults found in [lib/defaults.js](https://github.com/axios/axios/blob/master/lib/defaults.js#L28), then `defaults` property of the instance, and finally `config` argument for the request. The latter will take precedence over the former. Here's an example.
+The config will be merged with the following order of precedence:
+1. Library defaults found in [lib/defaults.js](https://github.com/axios/axios/blob/master/lib/defaults.js#L28)
+2. The `defaults` property of the instance
+3. The `config` argument for the request.
+
+Passing the `config` argument will completely **override** what is specified in the `defaults` property for some keys. These are currently `url`, `method`, `params`, and `data`.
```

Comment:
That's what I want to emphasize for users. We have different merging strategies for different fields, i.e.

- always from request config, i.e. url/method
- deep merge, i.e. headers
- merge by replacement, i.e. other fields

And I also want to mention something that will affect the final config. Including but no limiting for,

- relationships between config.header.common/[method] and config[header]
- the role of default `config.transformRequest`
- interceptors